src/ariel-os-embassy-common/src/gnss.rs

```diff
@@ -0,0 +1,119 @@
+//! Common types for GNSS functionality across different HALs.
+use embassy_sync::{
+    blocking_mutex::raw::CriticalSectionRawMutex,
+    watch::{Receiver, Sender, Watch},
+};
+
+/// Maximum number of concurrent receivers for GNSS data.
+pub const MAX_WATCH_RECEIVERS: usize = 4;
+/// Embassy Watch to hold the latest GNSS data and notify receivers.
+pub type GnssDataWatch = Watch<CriticalSectionRawMutex, GnssData, MAX_WATCH_RECEIVERS>;
+/// Type alias for the receiver of GNSS data (receiver of `GnssDataWatch`).
+pub type GnssDataReceiver<'a> =
+    Receiver<'a, CriticalSectionRawMutex, GnssData, MAX_WATCH_RECEIVERS>;
+/// Type alias for the sender of GNSS data (sender of `GnssDataWatch`).
+#[doc(hidden)]
+pub type GnssDataSender<'a> = Sender<'a, CriticalSectionRawMutex, GnssData, MAX_WATCH_RECEIVERS>;
+
+/// Operation modes for the GNSS module.
+#[derive(Clone, Copy)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub enum GnssOperationMode {
+    /// Always keep the GNSS module active.
+    Continuous,
+    /// Update the GNSS fix periodically.
+    Periodic(u16), // Period in seconds
+    /// Try to get a GNSS fix only when requested, you won't be able to get updates using `get_receiver`.
+    SingleShot(u16), // Timeout in seconds
+}
+
+/// Configuration for the GNSS.
+///
+/// You can customize it using the `gnss-config-override` feature.
+#[derive(Clone, Copy)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub struct Config {
+    /// The mode of GNSS to use.
+    pub operation_mode: GnssOperationMode,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            operation_mode: GnssOperationMode::Continuous,
+        }
+    }
+}
+
+/// Represents position data from GNSS.
+#[derive(Debug, Clone)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub struct GnssPosition {
+    /// Latitude in degrees
```

Comment:
All doc comments should end with a period.
```suggestion
    /// Latitude in degrees.
```